docs/source/caching.rst

```diff
@@ -0,0 +1,270 @@
+=================
+Caching
+=================
+
+Resolve Caching
+===============
+
+Resolve caching is a feature that caches resolves to a memcached (in-memory) server. Because the server is in-memory,
```

Comment:
```suggestion
Resolve caching is a feature that caches resolves to a `memcached <https://memcached.org/>`_ (in-memory) server. Because the server is in-memory,
```
I think we can probably merge the "benefits" and downsides" into this section. Also the server isn't in-memory. It's the data it holds that is in memory. I would also mention that memcached is a widely used, easy to deploy (because there is no storage needed and because it's a single process/executable) and simple server.